spring-integration-sftp/src/main/java/org/springframework/integration/sftp/session/SftpSession.java

```diff
@@ -278,4 +278,20 @@ public ChannelSftp getClientInstance() {
 		return this.channel;
 	}
 
+	@Override
+	public boolean test() {
+		return isOpen() && doTest();
+	}
+
+	private boolean doTest() {
+		try {
+			this.channel.stat(this.channel.getHome());
```

Comment:
Why `jschSession.sendKeepAliveMsg()` doesn't work for us?

Reply:
That command doesn't wait for a reply so we get false positives (e.g. when pulling the network cable).